tools/mcms_proposal/utils.go

```diff
@@ -0,0 +1,76 @@
+package mcms_proposal
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/ccip-owner-contracts/tools/configwrappers"
+	"github.com/smartcontractkit/ccip-owner-contracts/tools/gethwrappers"
+)
+
+type ContractDeployBackend interface {
+	bind.ContractBackend
+	bind.DeployBackend
+}
+
+func mapMCMAddresses(metadatas map[string]ChainMetadata) map[string]common.Address {
+	m := make(map[string]common.Address)
+	for k, v := range metadatas {
+		m[k] = v.MCMAddress
+	}
+	return m
+}
+
+func mapSignatures(signatures []Signature) []gethwrappers.ManyChainMultiSigSignature {
```

Comment:
yeah no preference on the name really, I'll change it to one of the suggestions you had. Not sure about sorting here, because I'm not sure if this function will be used elsewhere other than SetRoot. If its only used in SetRoot, it doesn't matter, but if it is gonna be used elsewhere, I wanna avoid adding in the complexity of sorting in places where its not relevant to the library consumers. sorting is only relevant when broadcasting onchain so I think we should confine that logic to that function